core/domain/suggestion_jobs_one_off.py

```diff
@@ -0,0 +1,149 @@
+                    written_translation_html = (
+                        written_translations_dict['translations_mapping'][
+                            content_id][language_code]['html'])
+                    html_string_list.append(written_translation_html)
+        html_string = ''.join(html_string_list)
+        soup = bs4.BeautifulSoup(
+            html_string.encode(encoding='utf-8'), 'html.parser')
+        if soup.findAll(name='oppia-noninteractive-math'):
+            yield ('Suggestion with Math', item.id)
+
+    @staticmethod
+    def reduce(key, values):
+        yield ('%d suggestions have Math RTEs in them with IDs.' % len(values),
+               values)
+
+
+class SuggestionMathMigrationOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """A one-time job that can be used to migrate the Math RTEs in the
+    suggestions to the new Math Schema.
+    """
+    _ERROR_KEY = 'validation_error'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        suggestion = suggestion_services.get_suggestion_by_id(item.id)
+        try:
+            suggestion.validate()
+        except Exception as e:
+            logging.error(
+                'Suggestion %s failed validation: %s' % (item.id, e))
+            yield (
+                SuggestionMathMigrationOneOffJob._ERROR_KEY,
+                'Suggestion %s failed validation: %s' % (item.id, e))
+            return
+        if (suggestion.suggestion_type ==
+                suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT):
+            suggestion_dict = suggestion.to_dict()
+            suggestion_dict['change']['content_html'] = (
+                html_validation_service.add_math_content_to_math_rte_components(
+                    suggestion_dict['change']['content_html']))
+            suggestion_dict['change']['translation_html'] = (
+                html_validation_service.add_math_content_to_math_rte_components(
+                    suggestion_dict['change']['translation_html']))
+            suggestion.change = (
+                exp_domain.ExplorationChange(suggestion_dict['change']))
+        elif (suggestion.suggestion_type ==
+              suggestion_models.SUGGESTION_TYPE_ADD_QUESTION):
+            suggestion_dict = suggestion.to_dict()
+            suggestion_dict['change']['question_dict'][
+                'question_state_data'] = (
+                    state_domain.State.convert_html_fields_in_state(
+                        suggestion_dict['change']['question_dict'][
+                            'question_state_data'],
+                        html_validation_service.
+                        add_math_content_to_math_rte_components))
+            suggestion.change = (
+                question_domain.QuestionSuggestionChange(
+                    suggestion_dict['change']))
+        suggestion.validate()
```

Comment:
Should we add validation in try-catch block and yeild for invalid cases?

Reply:
done